find_similar.py

```diff
@@ -0,0 +1,155 @@
+import torch
+import pyodbc
+from scipy.spatial.distance import cdist, cosine
+import numpy as np
+import json
+import chess
+import re
+
+import settings
+import model
+from inference import Inference
+
+
+def key(i):
+    return i[1]
+
+def l1_loss(matrix,target):
+    return [np.sum(np.abs(np.array(m)-np.array(target))) for m in matrix]
+    
+def nearest(matrix,target):
+    return cdist(matrix, np.atleast_2d([target]))
+    
+def cosine_dist(matrix,target):
+    return [cosine(m, target) for m in matrix]
+    
+similarity_functions = {"Closest vectors":nearest,"L1 Loos":l1_loss,"Cosine distance":cosine_dist}
+
+def get_move(line):
+    line = str(line)
+    game = []  # list of moves in pgn notation
+    for move in re.findall(
+        r"[^{\[.}\]]+ ", line.replace("?", "").replace("!", "")
+    ):  # Extract all moves without comments
+        game.extend(
+            move.strip().split(" ")
+        )  # Sometimes one string contains two moves with space between
+    return game
+
+
+def find_lowest(array, num, key=key):

```

Comment:
I don't like the key function. If it's used only here, use `lambda` to define inline function (e.g. `lambda item: item[1]`) - because the `key` function without context  was hard to understand

Reply:
Right